src/Clockify/ClockifyHandler.cs

```diff
@@ -125,6 +128,12 @@ public class ClockifyHandler : IBotHandler
             try
             {
                 await _clockifyService.GetCurrentUserAsync(userProfile.ClockifyToken);
+                // It will be removed when only ClockifyTokenId will be used
```

Comment:
I thought we'd do something like this, which would automatically put us where we want to be (_all_ tokens present in KV _only_):

```csharp
            switch (userProfile.ClockifyTokenId)
            {
                case null when userProfile.ClockifyToken == null:
                    // first sign in
                    await _clockifySetupDialog.RunAsync(turnContext, _dialogState, cancellationToken);
                    return true;
                case null when userProfile.ClockifyToken != null:
                    // old storage, migrating and cleaning
                    try
                    {
                        await _clockifyService.GetCurrentUserAsync(userProfile.ClockifyToken);
                        var tokenData = await _tokenRepository.WriteAsync(userProfile.ClockifyToken);
                        userProfile.ClockifyToken = null;
                        userProfile.ClockifyTokenId = tokenData.Id;
                        return false;
                    }
                    catch (ErrorResponseException)
                    {
                        // old login invalid, asking again
                        await _clockifySetupDialog.RunAsync(turnContext, _dialogState, cancellationToken);
                        return true;
                    }
                default:
                    try
                    {
                        var token = await _tokenRepository.ReadAsync(userProfile.ClockifyTokenId!);
                        await _clockifyService.GetCurrentUserAsync(token!.Value);
                        userProfile.ClockifyToken = null; // just to be sure :)
                        return false;
                    }
                    catch (ErrorResponseException)
                    {
                        // old login invalid, asking again
                        await _clockifySetupDialog.RunAsync(turnContext, _dialogState, cancellationToken);
                        return true;
                    }
            }
```

Of course we could write it down more elegantly if needed

Reply:
I'm also thinking that now we've complicated things by introducing KV, paying up the cost of token validation every time it's damn stupid. We could - at this stage - accept any non-null token, so no call to the KV is needed. If we catch an unauthorized exception during any dialog due to the token, we flag the user (via a `userProfile` field, for example. a `tokenExpired`) and at the next iteration we prompt them for a new valid one.

This way we reduce time and costs of communicating to the KV one time uselessly at every message

Reply:
I prefer classic if/else, because one of the cases will be removed.